test/unit/dialects/oracledb.js

```diff
@@ -39,4 +36,56 @@ describe("OracleDb externalAuth", function() {
       expect(spy).to.have.callCount(1);
       expect(spy).to.have.been.calledWith(connectionWithExternalAuth);
   });
+
+  after(function() {
+    sinon.restore(knexInstance.client.driver.getConnection);
+  });
+
+});
+
+describe("OracleDb parameters", function() {
+  var knexClient;
+
+  before(function(done) {
+    var conf = _.clone(config.oracledb);
+    conf.fetchAsString = [ 'number', 'DATE', 'cLOb'];
+    knexClient = knex(conf);
+    knexClient.schema.createTable('fetchAsStringTable', function (table) {
+      table.increments();
+      table.float('testfloat');
+      table.date('testdate');
+    }).then(function() {
+      knexClient('fetchAsStringTable').insert({
+        testfloat: 7.32,
+        testdate: new Date(2017, 5, 7)
+      }).then(function() {
+        done();
+      }).catch(done);
+    }).catch(done);
+  });
+
+  it('fetchAsString parameter should return strings on selected types', function(done) {
+    knexClient('fetchAsStringTable').select().then(function(result) {
+      expect(result[0]).to.be.ok;
+      expect(result[0].testfloat).to.be.a('string');
+      expect(result[0].testdate).to.be.a('string');
+      knexClient.destroy(done);
```

Comment:
destroying client here makes all of these tests dependent to each other... sadly thats pretty much how all tests of knex are written.